src/Microsoft.DotNet.Tools.Compiler/Program.cs

```diff
@@ -236,6 +236,49 @@ private static bool CompileProject(ProjectContext context, CompilerCommandApp ar
                 }
 
                 compilerArgs.AddRange(dependency.SourceReferences);
+
+                foreach (var contentFile in dependency.ContentFiles)
+                {
+                    var path = contentFile.Path;
+                    if (contentFile.Preprocess)
+                    {
+                        var newPath = Path.Combine(intermediateOutputPath, Path.GetFileName(path));
+
+                        using (var input = File.OpenRead(path))
+                        {
+                            using (var output = File.Create(newPath))
+                            {
+                                PPFilePreprocessor.Preprocess(input, output, new Dictionary<string, string>()
+                                {
+                                    {"rootnamespace", context.ProjectFile.Name },
+                                    {"assemblyname", context.ProjectFile.Name }
+                                });
+                            }
+                        }
+
+                        path = newPath;
+                    }
+
+                    if (contentFile.CopyToOutput)
+                    {
+                        var destinationFile = Path.Combine(outputPath, contentFile.OutputPath);
+                        var destinationDirectory = Path.GetDirectoryName(destinationFile);
+                        if (!Directory.Exists(destinationDirectory))
+                        {
+                            Directory.CreateDirectory(destinationDirectory);
+                        }
+                        File.Copy(path, destinationFile, true);
+                    }
+
+                    if (contentFile.BuildAction == BuildAction.Compile)
+                    {
+                        compilerArgs.Add(path);
```

Comment:
The compiler args are going to come out sorta weird. Maybe we should do these in groups? GroupBy build action?